core/ledger/kvledger/kv_ledger_provider.go

```diff
@@ -302,11 +334,12 @@ func (p *Provider) openInternal(ledgerID string) (ledger.PeerLedger, error) {
 		}
 	}
 
-	// Create a kvLedger for this chain/ledger, which encapsulates the underlying data stores
-	// (id store, blockstore, state database, history database)
+	// TODO: We have way more function arguments and it makes the code look ugly.
+	// Need to reduce it - FAB-17683
```

Comment:
Done. Removed all **new** TODOs. 

We pass 8 variables present in the `Provider` struct individually instead of passing the `Provider` itself. By passing the provider to `newKVLedger()`, we can reduce the number of arguments from 13 to 6. It will be done in FAB-17683.